src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -475,10 +475,10 @@ private ClusterDescription ProcessReplicaSetChange(ClusterDescription clusterDes
                         foreach (var currentPrimary in currentPrimaries)
                         {
                             // kick off the server to invalidate itself
-                            currentPrimary.Invalidate();
+                            currentPrimary.Invalidate("NoLongerPrimary");
```

Comment:
Not sure what happens here, is it possible to be in this code path when the server is still primary?

Reply:
I had to think about this one too...

Here's the conclusion I came to: the old primary is no longer primary but we haven't noticed yet because we haven't yet done a heartbeat for the old primary, but the heartbeat for the new primary implies that the old primary is no longer the primary.

Reply:
Ack